src/odemis/util/peak.py

```diff
@@ -247,7 +247,7 @@ def _DoFit(self, future, spectrum, wavelength, type='gaussian'):
                 # Increase window size until peak detection finds enough peaks to fit
                 # the spectrum curve
                 peaks = Detect(smoothed, wavelength, lookahead=window_size, delta=5)[0]
-                if peaks == []:
+                if not peaks:
```

Comment:
That's *probably* correct. You need to check `Detect` never returns `None`.

Reply:
How would you test that?

Reply:
Read the code of `Detect()`

Reply:
Well, I did that and do not see any obvious part where it will return None. It can return an empty list, which is fine. But do not know if there is maybe a None in the list, which I do not know if relevant. But just want to be sure I do not miss something...

Reply:
If there is a None in the list, that's not an empty list. So it'll do the same thing in both way you write this `if`. So it's fine :-)